packages/cardano-services/src/Blockfrost/BlockfrostCache.ts

```diff
@@ -0,0 +1,86 @@
+import { AvailableNetworks } from '..';
+import { BlockFrostAPI } from '@blockfrost/blockfrost-js';
+import { BlockfrostCacheBuilder } from './builder';
+import { Logger } from 'ts-log';
+import { Pool } from 'pg';
+import { RunnableModule } from '@cardano-sdk/util';
+
+type BlockfrostMetrics = Awaited<ReturnType<BlockFrostAPI['poolsById']>>;
+
+export interface BlockfrostCacheConfig {
+  blockfrostApiKey: string;
+  cacheTtl: number;
+  network: AvailableNetworks;
+}
+
+export interface BlockfrostCacheDependencies {
+  db: Pool;
+  logger: Logger;
+}
+
+export class BlockfrostCache extends RunnableModule {
+  #api: BlockFrostAPI;
+  #builder: BlockfrostCacheBuilder;
+  #cacheTtl: number;
+  #db: Pool;
+  #shuttingDown?: boolean;
+
+  constructor(cfg: BlockfrostCacheConfig, deps: BlockfrostCacheDependencies) {
+    const { blockfrostApiKey, cacheTtl, network } = cfg;
+    const { db, logger } = deps;
+
+    super('blockfrost-cache', logger);
+
+    this.#api = new BlockFrostAPI({ network, projectId: blockfrostApiKey });
+    this.#builder = new BlockfrostCacheBuilder(db, logger);
+    this.#cacheTtl = cacheTtl;
+    this.#db = db;
+  }
+
+  protected initializeImpl() {
+    return Promise.resolve();
+  }
+
+  protected startImpl() {
+    return Promise.resolve();
```

Comment:
Seems this doesn't need to be a `RunnableModule`?

Reply:
Not yet, the intention is to check at the beginning of iteration in the main loop if a shutdown was requested to immediately exit rather than completing to refresh all the pools